Add funding to the readme and add funding file (#2372)
WordPress/WordPress-Coding-Standards: README.md

* [Fixing errors or ignoring them](#fixing-errors-or-ignoring-them)
    + [Tools shipped with WordPressCS](#tools-shipped-with-wordpresscs)
* [Contributing](#contributing)
* [Funding](#funding)
* [License](#license)

---

## Introduction

This project is a collection of [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) rules (sniffs) to validate code developed for WordPress. It ensures code quality and adherence to coding conventions, especially the official [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).

> This project needs funding. If you want to donate read [here](#funding) about how you can help.

Reviewer: ```suggestion
This project needs funding. [Find out how you can help](#funding).
```

* [Contributing](#contributing)
* [License](#license)

---

## This project needs funding

> WordPressCS 3.0.0 has costs thousands of hours of work and the vast majority of work has been done by one, mostly unpaid, contributor (@jrfnl), with code review support from two fellow maintainers.

Reviewer 2: ```suggestion
Maintaining WordPressCS costs thousands of hours of work and the vast majority of work until now has been done by one, mostly unpaid, contributor (@jrfnl), with code review support from two fellow maintainers.
```

**This situation is unsustainable.**

> A large part of the WordPress community, including WordPress Core, relies heavily on the WordPress Coding Standards for code quality and security checks and while the community has been pretty vocal with copious complaints about the delayed release, barely anyone has stepped up and actually contributed.

Reviewer 2: ```suggestion
A large part of the WordPress community, including WordPress Core, relies heavily on the WordPress Coding Standards for code quality and security checks.
```

(That bit is specific to the WPCS 3.0 release)

Unless funding is found to continue maintaining WordPressCS and its dependencies, the future is bleak and maintenance will be halted.

> If you are an agency, a corporate entity or just a passionate user who wishes to help to fund the continued maintenance of the WordPress CS, you can donate to the [WP PHP Open Collective](https://opencollective.com//thewpcc/contribute/wp-php-63406).

Reviewer 2: ```suggestion
If you are an agency, a corporate entity or even a passionate user who wishes to help fund the continued maintenance of WordPressCS, you can donate to the [WP PHP Open Collective](https://opencollective.com//thewpcc/contribute/wp-php-63406).
```
